### PR TITLE
Bug 2034484: feat: updated library-go

### PR DESCRIPTION
**Pull request by eggfoobar**
Bumping the library-go dependency to latest, this should take advantage of leader election changes for SNO clusters proposed in this [library-go PR](https://github.com/openshift/library-go/pull/1242) and performance improvements in this [PR](https://github.com/openshift/library-go/pull/1235).

Changes:
  - updated library-go dep to latest
  - updated start command for the controllers to use library-go defaults for leader election

Signed-off-by: ehila <ehila@redhat.com>

### CODE REVIEW

File: pkg/operator/sync.go [hunk 1]
```diff
@@ -194,10 +195,12 @@ func (optr *Operator) syncWebhookConfiguration() error {
 }
 
 func (optr *Operator) syncValidatingWebhook() error {
-	expectedGeneration := resourcemerge.ExpectedValidatingWebhooksConfiguration(mapiwebhooks.NewValidatingWebhookConfiguration().Name, optr.generations)
```
Comment (eggfoobar) [inline pkg/operator/sync.go:197]: I'm not 100% sure on the ramifications of this change here, would love some input and context. The `expectedGeneration` was removed from this methods signature per this [PR](https://github.com/openshift/library-go/pull/1256). Any potential problems that we might need to consider here?

**Comment by eggfoobar** [comment]
/retitle Bug 2034484: feat: updated library-go
This PR bumps to latest library-go to utilize changes for performance gains on SNO DU clusters

File: cmd/machine-api-operator/start.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"strconv"
 
 	osconfigv1 "github.com/openshift/api/config/v1"
+
```
Comment (JoelSpeed) [inline cmd/machine-api-operator/start.go:12]: This new line seems to be randomly added, can we remove it please?

**Review by JoelSpeed — COMMENTED** [review]
/approve

File: cmd/machine-api-operator/start.go (hunk 2 above)
Comment (eggfoobar) [reply cmd/machine-api-operator/start.go]: Sure thing, updated

**Pull request merged**